ror-tools-core/src/main/scala/tech/beshu/ror/tools/core/patches/base/EsPatchExecutor.scala

```diff
@@ -0,0 +1,168 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.tools.core.patches.base
+
+import tech.beshu.ror.tools.core.patches.base.EsPatchExecutor.EsPatchStatus
+import tech.beshu.ror.tools.core.patches.base.EsPatchExecutor.EsPatchStatus.*
+import tech.beshu.ror.tools.core.patches.internal.RorPluginDirectory
+import tech.beshu.ror.tools.core.utils.RorToolsError.*
+import tech.beshu.ror.tools.core.utils.{EsDirectory, FileUtils, InOut, RorToolsError}
+
+import scala.util.{Failure, Success, Try}
+
+final class EsPatchExecutor(rorPluginDirectory: RorPluginDirectory,
```

Comment:
Let's consider the ES upgrade.

Let's say: 
1. user has ES 8.0.0, installs ROR 1.63.0 and patches it
2. user upgrades the ES to 9.0.0 without unpatching and ROR
3. user starts the new ES

Is ROR able to detect that the installation is broken?

Reply:
That situation can be detected, if the jar files in /lib directory have changed during the update (and according to some sources on the Internet, the /lib directory is replaced during upgrade; but I guess it depends on how the ES is installed and how it is upgraded)

Reply:
Ok, great. Are we able to simulate this in our tests?

Reply:
I added explicit handling of ES version in the plugin. The metadata file now looks something like that 

```
{
  "rorVersion" : "1.64.0-pre6",
  "esVersion" : "9.0.0",
  "patchedFilesMetadata" : [
    {
      "path" : "/tmp/3483890482512061106/es/lib/elasticsearch-9.0.0.jar",
      "hash" : "2d115f55b9befc1381284756292901af10ec7b28b90eecd5350ea08d613ba92a"
    }
  ]
}
```

And the error message for this case is ```The patch was performed on Elasticsearch version 1.2.3, but currently installed version is 9.0.0. It should be unpatched using ROR for the ES version used for patching and patched again. For patching instructions see our docs: https://docs.readonlyrest.com/elasticsearch#id-3.-patch-elasticsearch```

Reply:
```As a result, the Elasticsearch is in a corrupted state. ES must be reinstalled.```